quil-rs/src/expression/mod.rs

```diff
@@ -135,15 +135,15 @@ impl Hash for Expression {
                 "Infix".hash(state);
                 operator.hash(state);
                 match operator {
-                    InfixOperator::Plus | InfixOperator::Star => {
-                        // commutative, so put left & right in decreasing order by hash value
-                        let (a, b) = (
-                            min_by_key(&left, &right, hash_to_u64),
-                            max_by_key(&left, &right, hash_to_u64),
-                        );
-                        a.hash(state);
-                        b.hash(state);
-                    }
+                    //InfixOperator::Plus | InfixOperator::Star => {
+                    //    // commutative, so put left & right in decreasing order by hash value
```

Comment:
I agree that this is costly, but we put it in for #27. Are we deciding to no longer ensure that, e.g., `1 + 2 == 2 + 1`?

Reply:
Alternatively, should we just derive `Hash`, `PartialEq,` and `Eq` and call it a day?

Reply:
In fact, I think this is the way to go. #27 only handles commutativity, but not other rules that lead to equality between expressions. Since we're simplifying expressions with these rules, enforcing some but not all in the hashing & equality implementations seems wrong and wasteful. I'll open a new PR.